pkg/block/azure/walker.go

```diff
@@ -104,6 +117,34 @@ func (a *BlobWalker) Walk(ctx context.Context, storageURI *url.URL, op block.Wal
 	return nil
 }
 
+// isBlobItemFolder returns true if the blob item is a folder.
+// Make sure that metadata is populated before calling this function.
```

Comment:
Can you expand in the comment exactly what is needed to populate the metadata

Reply:
Adding list example - I think it is too much as the caller can get or produce the information in many ways.